src/redux/contacts.ts

```diff
@@ -1,13 +1,74 @@
 import { omit, toLower } from 'lodash';
 import { getContacts, saveContacts } from '../handlers/localstorage/contacts';
+import { AppDispatch, AppGetState } from './store';
+import { Network } from '@rainbow-me/helpers/networkTypes';
 
 // -- Constants --------------------------------------- //
 const CONTACTS_UPDATE = 'contacts/CONTACTS_UPDATE';
 const CONTACTS_LOAD = 'contacts/CONTACTS_LOAD';
 const CONTACTS_CLEAR_STATE = 'contacts/CONTACTS_CLEAR_STATE';
 
+// -- Types ------------------------------------------ //
+
+/**
+ * A contact's information.
+ */
+export interface Contact {
+  /**
+   * The contact's address.
+   */
+  address: string;
+
+  /**
+   * A color index, based on the values in `avatars` in
+   * `src/utils/profileUtils.ts`.
+   */
+  color: number;
+
+  /**
+   * The network.
+   */
+  network: Network;
+
+  /**
+   * The contact's nickname.
+   */
+  nickname: string;
+}
+
+/**
+ * The `contacts` reducer's state.
+ */
+interface ContactsState {
+  contacts: {
+    [address: string]: Contact;
+  };
+}
+
+/**
+ * An action for the `contacts` reducer.
+ */
+type ContactsAction =
+  | ContactsUpdateAction
+  | ContactsLoadAction
+  | ContactsClearStateAction;
+
+interface ContactsUpdateAction {
+  type: typeof CONTACTS_UPDATE;
+  payload: ContactsState['contacts'];
+}
+
+interface ContactsLoadAction {
+  type: typeof CONTACTS_LOAD;
+  payload: ContactsState['contacts'];
+}
+
+interface ContactsClearStateAction {
+  type: typeof CONTACTS_CLEAR_STATE;
+}
+
 // -- Actions ---------------------------------------- //
-export const contactsLoadState = () => async dispatch => {
+export const contactsLoadState = () => async (dispatch: AppDispatch) => {
```

Comment:
Haven't dug into why exactly, but the `AppDispatch` type is actually resolving to `Dispatch<any>` so there's not any type checking happening on any `dispatch` calls in these files. Might be worth explicitly typing them like this: `Dispatch< ContactsAction>` — when I do this it surfaces type errors that were previously being masked.